File: coins/src/main/java/com/bubelov/coins/dagger/Injector.java
package com.bubelov.coins.dagger;

/**
 * Author: Igor Bubelov
 * Date: 26/03/16 17:59
 */

public enum Injector {
    INSTANCE;

    AppComponent appComponent;

    public AppComponent getAppComponent() {
        if (appComponent == null) {
            appComponent = DaggerAppComponent.create();
        }

        return appComponent;
    }
}

File: app/src/main/java/com/bubelov/coins/di/Injector.kt
package com.bubelov.coins.di

import android.content.Context

/**
 * @author Igor Bubelov
 */

object Injector {
    lateinit var appComponent: AppComponent
    private set

    fun init(context: Context) {
        appComponent = DaggerAppComponent.builder().context(context).build()
    }
}